lib/THC/THCGeneral.c

```diff
@@ -212,15 +341,33 @@ void THCState_setStream(THCState *state, int device, int stream)
     THError("%d is not a device", device + 1 /* back to Torch index */);
   }
 
-  /* Stream 0 is the default stream, 1 ... `numUserStreams` are Torch streams */
   if (stream > state->numUserStreams || stream < 0)
   {
     THError("%d is not a stream", stream);
   }
-
-  state->currentStream = state->streamsPerDevice[device][stream];
+  state->currentStream =
+    THCState_getDeviceStream(state, device, stream);
   state->currentPerDeviceStream = stream;
-  THCudaBlas_setStream(state, device, state->currentStream);
+  THCublasCheck(cublasSetStream(state->currentBlasHandle,
+                                state->currentStream));
+}
+
+void THCState_setBlasHandle(THCState *state, int device, int handle)
```

Comment:
You only really want to call this method if `device` is the current device, right? Otherwise `currentBlasHandle` holds a handle for a device other than the one we're currently using.